anesthetic/samples.py

```diff
@@ -482,6 +482,102 @@ class MCMCSamples(Samples):
     def _constructor(self):
         return MCMCSamples
 
+    def remove_burn_in(self, burn_in, reset_index=False, inplace=False):
+        """Remove burn-in samples from each MCMC chain.
+
+        Parameters
+        ----------
+        burn_in: int or float or array_like
+            if 0 < burn_in < 1: remove first ``burn_in`` fraction of samples
+            elif 1 < burn_in: remove first ``burn_in`` number of samples
+            elif type(burn_in)==list: specify different burn-in for each chain
+
+        reset_index: bool, default=False
+            Whether to reset the index counter to start at zero or not.
+
+        inplace: bool, default=False
+            Indicates whether to modify the existing array or return a copy.
+
+        """
+        chains = self.groupby(('chain', '$n_\\mathrm{chain}$'),
+                              group_keys=False)
+        nchains = chains.ngroups
+        if isinstance(burn_in, (int, float)) and 0 < burn_in < 1:
+            nsamples = chains.count().iloc[:, 0].to_numpy()
+            ndrop = np.ceil(burn_in * nsamples).astype(int)
+        elif isinstance(burn_in, (int, float)):
+            ndrop = np.full(nchains, burn_in, dtype=int)
+        elif isinstance(burn_in, (list, tuple, np.ndarray)) \
+                and len(burn_in) == nchains:
+            ndrop = burn_in
+        else:
+            raise ValueError("`burn_in` has to be a positive scalar or an "
+                             "array of length matching the number of chains "
+                             "`nchains=%d`. However, you provided "
+                             "`burn_in=%s`" % (nchains, burn_in))
+        data = self.drop(chains.apply(lambda g: g.head(ndrop[g.name-1])).index,
+                         inplace=inplace)
+        if reset_index:
+            data = data.reset_index(drop=True, inplace=inplace)
+        return data
+
+    def Gelman_Rubin(self, params=None):
+        """Gelman--Rubin convergence statistic of multiple MCMC chains.
+
+        Determine the Gelman--Rubin convergence statistic ``R-1`` by computing
+        and comparing the within-chain variance and the between-chain variance.
+        This follows the routine as outlined in Lewis (2013), section
+        IV.A., https://arxiv.org/abs/1304.4473.
+
+        Note that this requires more than one chain. To circumvent this, you
+        could overwrite the ``'chain'`` column, splitting the samples into two
+        or more sets.
+
+        Parameters
+        ----------
+        params: list(str)
+            List of column names (i.e. parameters) to be included in the
+            convergence calculation.
+            Default: all parameters (except those parameters that contain
+            'prior', 'chi2', or 'logL' in their names)
```

Comment:
This by default would include derived parameters -- is this how cobaya does it?

Reply:
I believe so, and I don't see a reason why they shouldn't be included.